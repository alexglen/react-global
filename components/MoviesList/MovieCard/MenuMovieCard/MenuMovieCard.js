import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { typeEdit } from '../../../../constants';
import {
  openCardModal,
  openDeleteModal,
  setCurrentCardId,
  setTypeEvent,
} from '../../../../redux/actions';
import styles from './MenuMovieCard.module.scss';

const MenuMovieCard = ({ isMenuModalOpen, setIsMenuModalOpen, movieCardId }) => {
  const dispatch = useDispatch();

  const editMovie = () => {
    dispatch(openCardModal());
    dispatch(setCurrentCardId(movieCardId));
    setIsMenuModalOpen(false);
    dispatch(setTypeEvent(typeEdit));
  };

  const deleteMovie = () => {
    dispatch(openDeleteModal());
    dispatch(setCurrentCardId(movieCardId));
    setIsMenuModalOpen(false);
  };

  const close = () => {
    setIsMenuModalOpen(false);
    dispatch(setCurrentCardId(movieCardId));
  };

  const menu = isMenuModalOpen && (
    <div className={styles.menu}>
      <ul>
        <li onClick={editMovie}>Edit</li>
        <li onClick={deleteMovie}>Delete</li>
      </ul>
      <div className={styles.close} onClick={close} />
    </div>
  );

  return menu;
};

MenuMovieCard.propTypes = {
  isMenuModalOpen: PropTypes.bool,
  setIsMenuModalOpen: PropTypes.func,
  movieCardId: PropTypes.string,
};

export default MenuMovieCard;
