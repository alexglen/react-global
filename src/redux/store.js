import { createStore, applyMiddleware, combineReducers, compose } from "redux";
import thunk from "redux-thunk";
import { filterReducer } from "./reducers/filterReducer";
import { modalsReducer } from "./reducers/modalsReducer";
import { moviesReducer } from "./reducers/moviesReducer";
import { searchReducer } from "./reducers/searchReducer";
import { sortingReducer } from "./reducers/sortingReducer";

const reducer = combineReducers({
  moviesReducer,
  filterReducer,
  sortingReducer,
  searchReducer,
  modalsReducer,
});

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(reducer, composeEnhancers(applyMiddleware(thunk)));

export default store;
