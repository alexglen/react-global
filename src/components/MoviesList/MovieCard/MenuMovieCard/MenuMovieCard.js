import React, { useContext } from 'react';
import { StatusModalsContext } from '../../../../context/StatusModalsContext';
import { EDIT } from '../../../../constants';
import PropTypes from 'prop-types';
import './MenuMovieCard.scss';

const MenuMovieCard = ({ isMenuModalOpen, setIsMenuModalOpen, movieCardId }) => {
  const { setIsDeleteModalOpen, setIsCardModalOpen, setIdChosenCard, setTypeOfEvent } = useContext(
    StatusModalsContext
  );

  const editMovie = () => {
    setIsCardModalOpen(true);
    setIdChosenCard(movieCardId);
    setIsMenuModalOpen(false);
    setTypeOfEvent(EDIT);
  };

  const deleteMovie = () => {
    setIsDeleteModalOpen(true);
    setIdChosenCard(movieCardId);
    setIsMenuModalOpen(false);
  };

  const menu = isMenuModalOpen && (
    <div className='menu-modal'>
      <ul>
        <li onClick={editMovie}>Edit</li>
        <li onClick={deleteMovie}>Delete</li>
      </ul>
      <div className='close' onClick={() => setIsMenuModalOpen(false)} />
    </div>
  );

  return menu;
};

MenuMovieCard.propTypes = {
  isMenuModalOpen: PropTypes.bool,
  setIsMenuModalOpen: PropTypes.func,
  movieCardId: PropTypes.string,
};

export default MenuMovieCard;
