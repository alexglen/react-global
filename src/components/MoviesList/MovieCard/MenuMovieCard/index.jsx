import React from "react";
import { useDispatch } from "react-redux";
import { openCardModal, openDeleteModal, setCurrentCardId, setTypeEvent } from "../../../../redux/actions";
import { typeEdit } from "../../../../constants";
import PropTypes from "prop-types";
import "./MenuMovieCard.scss";

function MenuMovieCard({ isMenuModalOpen, setIsMenuModalOpen, movieCardId }) {
  const dispatch = useDispatch();

  const editMovie = () => {
    dispatch(openCardModal());
    dispatch(setCurrentCardId(movieCardId));
    setIsMenuModalOpen(false);
    dispatch(setTypeEvent(typeEdit));
  };

  const deleteMovie = () => {
    dispatch(openDeleteModal());
    dispatch(setCurrentCardId(movieCardId));
    setIsMenuModalOpen(false);
  };

  const close = () => {
    setIsMenuModalOpen(false);
    dispatch(setCurrentCardId(movieCardId));
  };

  const menu = isMenuModalOpen && (
    <div className="menu-modal">
      <ul>
        <li onClick={editMovie}>Edit</li>
        <li onClick={deleteMovie}>Delete</li>
      </ul>
      <div className="close" onClick={close} />
    </div>
  );

  return menu;
}

MenuMovieCard.propTypes = {
  isMenuModalOpen: PropTypes.bool,
  setIsMenuModalOpen: PropTypes.func,
  movieCardId: PropTypes.string,
};

export default MenuMovieCard;
