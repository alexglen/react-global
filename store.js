import { createStore, applyMiddleware, combineReducers, compose } from 'redux';
import thunk from 'redux-thunk';
import { filterReducer } from './reducers/filterReducer';
import { modalsReducer } from './reducers/modalsReducer';
import { moviesReducer } from './reducers/moviesReducer';
import { sortingReducer } from './reducers/sortingReducer';

const reducer = combineReducers({
  movies: moviesReducer,
  filters: filterReducer,
  sorting: sortingReducer,
  modals: modalsReducer,
});

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(reducer, composeEnhancers(applyMiddleware(thunk)));

export default store;
