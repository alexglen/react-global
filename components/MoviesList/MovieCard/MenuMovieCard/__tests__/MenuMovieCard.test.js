import { BrowserRouter } from "react-router-dom";
import store from "../../../../../redux/store";
import MenuMovieCard from "../MenuMovieCard";
import { create } from "react-test-renderer";
import { Provider } from "react-redux";

const getComponent = (isMenuModalOpen = false) => {
  return create(
    <Provider store={store}>
      <BrowserRouter>
        <MenuMovieCard isMenuModalOpen setIsMenuModalOpen={() => jest.fn()} movieCardId="-MWJn2JOvxRpVKrK0Ye3" />
      </BrowserRouter>
    </Provider>
  );
};

describe("MenuMovieCard was rendered", () => {
  test("render <MenuMovieCard />", () => {
    const component = getComponent();
    expect(component.toJSON()).toMatchSnapshot();
  });

  test("render <MenuMovieCard />", () => {
    const component = getComponent(true);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
