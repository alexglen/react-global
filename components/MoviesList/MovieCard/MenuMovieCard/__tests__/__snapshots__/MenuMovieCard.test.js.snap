// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuMovieCard was rendered render <MenuMovieCard /> 1`] = `
<div
  className="menu-modal"
>
  <ul>
    <li
      onClick={[Function]}
    >
      Edit
    </li>
    <li
      onClick={[Function]}
    >
      Delete
    </li>
  </ul>
  <div
    className="close"
    onClick={[Function]}
  />
</div>
`;

exports[`MenuMovieCard was rendered render <MenuMovieCard /> 2`] = `
<div
  className="menu-modal"
>
  <ul>
    <li
      onClick={[Function]}
    >
      Edit
    </li>
    <li
      onClick={[Function]}
    >
      Delete
    </li>
  </ul>
  <div
    className="close"
    onClick={[Function]}
  />
</div>
`;
